src/index/CompressedRelation.cpp

```diff
@@ -529,31 +527,30 @@ CompressedRelationReader::getBlocksForJoin(
 
 // _____________________________________________________________________________
 IdTable CompressedRelationReader::scan(
-    const ScanSpecification& scanSpec,
-    std::span<const CompressedBlockMetadata> blocks,
+    const ScanSpecAndBlocks& scanSpecAndBlocks,
     ColumnIndicesRef additionalColumns,
     const CancellationHandle& cancellationHandle,
     [[maybe_unused]] const LocatedTriplesPerBlock& locatedTriplesPerBlock,
     const LimitOffsetClause& limitOffset) const {
+  const auto& scanSpec = scanSpecAndBlocks.scanSpec_;
   auto columnIndices = prepareColumnIndices(scanSpec, additionalColumns);
   IdTable result(columnIndices.size(), allocator_);
-
   // Compute an upper bound for the size and reserve enough space in the
   // result.
-  auto relevantBlocks = getRelevantBlocks(scanSpec, blocks);
-  auto sizes =
-      relevantBlocks | ql::views::transform(&CompressedBlockMetadata::numRows_);
+  auto sizes = scanSpecAndBlocks.getBlockMetadataView() |
```

Comment:
same question here.
Does the ScanSpecAndBlocks constructor verify that it only gets blocks that are relevant for the scan? In that case I like the new change
(implement that check there).